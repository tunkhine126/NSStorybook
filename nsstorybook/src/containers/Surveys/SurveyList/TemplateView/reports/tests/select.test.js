/**
 *
 * Tests for Select
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import TestingComponent from 'components/shared/Testing';
import { render } from '@testing-library/react';

import Select from '../select';

describe('<Select />', () => {
  const props = {
    responses: [
      { sourceValue: 'Test1, Test2, Test3' },
      { sourceValue: 'Test3' },
      { sourceValue: 'Test1' },
      { sourceValue: 'Test2' },
    ],
    options: ['Test1', 'Test2', 'Test3'],
    optionLabels: {
      Test1: 'Test Options 1',
      Test2: 'Test Options 2',
      Test3: 'Test Options 3',
    },
  };

  const propsVersion2 = {
    responses: [{ sourceValue: 'Test4' }, { sourceValue: 'Test5' }],
    options: ['Test1', 'Test2', 'Test3'],
    optionLabels: {
      Test1: 'Test Options 1',
      Test2: 'Test Options 2',
      Test3: 'Test Options 3',
    },
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <Select {...props} />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <Select {...propsVersion2} />
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
