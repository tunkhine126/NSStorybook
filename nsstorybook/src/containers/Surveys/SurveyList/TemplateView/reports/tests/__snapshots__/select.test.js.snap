// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> Should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root makeStyles-chartContainer-179"
>
  <div
    class="MuiGrid-root makeStyles-container-300 MuiGrid-container MuiGrid-align-items-xs-center"
  >
    <div
      class="MuiGrid-root makeStyles-percentageContainer-301 MuiGrid-item"
    >
      <p
        class="MuiTypography-root makeStyles-percentageText-302 MuiTypography-body1"
      >
        0
        %
      </p>
    </div>
    <div
      class="MuiGrid-root makeStyles-barContainer-299 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-true"
    >
      <div
        class="makeStyles-percentageBar-303"
      />
      <div
        class="MuiGrid-root makeStyles-barTextContainer-298 MuiGrid-container MuiGrid-item MuiGrid-justify-xs-space-between"
      >
        <div
          class="MuiGrid-root makeStyles-responseContainer-304"
        >
          <p
            class="MuiTypography-root makeStyles-responseText-305 MuiTypography-body1"
          >
            Test Options 1
          </p>
        </div>
        <div
          class="MuiGrid-root"
        >
          <p
            class="MuiTypography-root makeStyles-numberOfResponses-306 MuiTypography-body1"
          >
            0
             
            Responses
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root makeStyles-container-339 MuiGrid-container MuiGrid-align-items-xs-center"
  >
    <div
      class="MuiGrid-root makeStyles-percentageContainer-340 MuiGrid-item"
    >
      <p
        class="MuiTypography-root makeStyles-percentageText-341 MuiTypography-body1"
      >
        0
        %
      </p>
    </div>
    <div
      class="MuiGrid-root makeStyles-barContainer-338 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-true"
    >
      <div
        class="makeStyles-percentageBar-342"
      />
      <div
        class="MuiGrid-root makeStyles-barTextContainer-337 MuiGrid-container MuiGrid-item MuiGrid-justify-xs-space-between"
      >
        <div
          class="MuiGrid-root makeStyles-responseContainer-343"
        >
          <p
            class="MuiTypography-root makeStyles-responseText-344 MuiTypography-body1"
          >
            Test Options 2
          </p>
        </div>
        <div
          class="MuiGrid-root"
        >
          <p
            class="MuiTypography-root makeStyles-numberOfResponses-345 MuiTypography-body1"
          >
            0
             
            Responses
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root makeStyles-container-348 MuiGrid-container MuiGrid-align-items-xs-center"
  >
    <div
      class="MuiGrid-root makeStyles-percentageContainer-349 MuiGrid-item"
    >
      <p
        class="MuiTypography-root makeStyles-percentageText-350 MuiTypography-body1"
      >
        0
        %
      </p>
    </div>
    <div
      class="MuiGrid-root makeStyles-barContainer-347 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-true"
    >
      <div
        class="makeStyles-percentageBar-351"
      />
      <div
        class="MuiGrid-root makeStyles-barTextContainer-346 MuiGrid-container MuiGrid-item MuiGrid-justify-xs-space-between"
      >
        <div
          class="MuiGrid-root makeStyles-responseContainer-352"
        >
          <p
            class="MuiTypography-root makeStyles-responseText-353 MuiTypography-body1"
          >
            Test Options 3
          </p>
        </div>
        <div
          class="MuiGrid-root"
        >
          <p
            class="MuiTypography-root makeStyles-numberOfResponses-354 MuiTypography-body1"
          >
            0
             
            Responses
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
